vck-initializer/vck-initializer/main.go

```diff
@@ -0,0 +1,256 @@
+					log.Printf("Required '%s' annotation missing; skipping vck initializing", annotation)
+					_, err := clientset.AppsV1beta1().Deployments(deployment.Namespace).Update(initializedDeployment)
+					if err != nil {
+						return err
+					}
+					return nil
+				}
+				//log.Print("annotation: ", a[annotation])
+				info := &data{}
+				err := json.Unmarshal([]byte(a[annotation]), info)
+				if err != nil {
+					return err
+				}
+				fmt.Println("Unmarshal:", info.MountPath)
+				vckVM, err := crdClient.VckV1().VolumeManagers(deployment.GetNamespace()).Get(info.Name, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				volumeVCK, affinityVCK, err := addVolumesAffinity(vckVM, info)
+				if err != nil {
+					return err
+				}
+				if info.Containers == nil {
+					for _, container := range deployment.Spec.Template.Spec.Containers {
+						info.Containers = append(info.Containers, container.Name)
+					}
+				}
+
+				for _, container := range info.Containers {
+					volumeMount, containerID, err := addVolumeMount(deployment, vckVM, container, info.MountPath)
+					if err != nil {
+						return err
+					}
+					initializedDeployment.Spec.Template.Spec.Containers[containerID].VolumeMounts = append(deployment.Spec.Template.Spec.Containers[containerID].VolumeMounts, *volumeMount)
+				}
+				initializedDeployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, *volumeVCK)
+				initializedDeployment.Spec.Template.Spec.Affinity = affinityVCK
+
+			}
+			oldData, err := json.Marshal(deployment)
+			if err != nil {
+				return err
+			}
+
+			newData, err := json.Marshal(initializedDeployment)
+			if err != nil {
+				return err
+			}
+
+			patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, v1beta1.Deployment{})
+			if err != nil {
+				return err
+			}
+
+			_, err = clientset.AppsV1beta1().Deployments(deployment.Namespace).Patch(deployment.Name, types.StrategicMergePatchType, patchBytes)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func addVolumesAffinity(vckVM *vckv1.VolumeManager, info *data) (*corev1.Volume, *corev1.Affinity, error) {
+	for _, item := range vckVM.Status.Volumes {
+		if info.ID == item.ID {
+			volumeVCK := corev1.Volume{
+				Name: "dataset-claim",
+				VolumeSource: corev1.VolumeSource{
+					HostPath: item.VolumeSource.HostPath,
+				},
+			}
+			affinityVCK := corev1.Affinity{
+				NodeAffinity: &item.NodeAffinity,
+			}
+			return &volumeVCK, &affinityVCK, nil
+		}
+	}
+	return nil, nil, errors.New("given id for vck does not exists")
+
+}
+
+func addVolumeMount(deployment *v1beta1.Deployment, vckVM *vckv1.VolumeManager, container string, mountPath string) (*corev1.VolumeMount, int, error) {
+	containerID := -1
+	for id, item := range deployment.Spec.Template.Spec.Containers {
+		if container == item.Name {
+			containerID = id
+			volumeMount := corev1.VolumeMount{
+				MountPath: mountPath,
+				Name:      "dataset-claim",
```

Comment:
You can pass in the volumeVCK and extract name from that instead of explicitly specifying it here.

Reply:
yes I agree, did that